### PR TITLE
cleancode: Make test logic simple

### PR DESCRIPTION
**Pull request by zhangxiaoyu-zidif**
**What this PR does / why we need it**:
Put return err forward, that make the code much readable.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/config/file_linux_test.go [hunk 1]
```diff
@@ -412,12 +412,13 @@ func writeFile(filename string, data []byte) error {
...
+	if err != nil {
+		return err
+	}
 	if err == nil && n < len(data) {
 		err = io.ErrShortWrite
 	}
-	if err1 := f.Close(); err == nil {
-		err = err1
-	}
+	err = f.Close()
```
Comment (derekwaynecarr) [inline pkg/kubelet/config/file_linux_test.go]: this may not get called if there was a write error.

File: pkg/kubelet/config/file_linux_test.go [hunk 2]
```diff
@@ -412,12 +412,13 @@ func writeFile(filename string, data []byte) error {
 		return err
 	}
 	n, err := f.Write(data)
+	if err != nil {
+		return err
+	}
 	if err == nil && n < len(data) {
```
Comment (derekwaynecarr) [inline pkg/kubelet/config/file_linux_test.go]: this err check is redundant with the earlier return.

**Review by derekwaynecarr — CHANGES_REQUESTED** [review]

**Comment by zhangxiaoyu-zidif** [comment]
@derekwaynecarr 
Thank you for review , sir.
I learn much from you. 
I change it. PTAL. =)

File: pkg/kubelet/config/file_linux_test.go [hunk 3]
```diff
@@ -411,13 +411,11 @@ func writeFile(filename string, data []byte) error {
 	if err != nil {
 		return err
 	}
-	n, err := f.Write(data)
+	n, _ := f.Write(data)
```
Comment (xiangpengzhao) [inline pkg/kubelet/config/file_linux_test.go]: Why ignoring the error of `f.Write`?

File: pkg/kubelet/config/file_linux_test.go [hunk 4]
```diff
@@ -411,13 +411,11 @@ func writeFile(filename string, data []byte) error {
...
 	if err == nil && n < len(data) {
 		err = io.ErrShortWrite
 	}
-	if err1 := f.Close(); err == nil {
-		err = err1
-	}
+	f.Close()
```
Comment (xiangpengzhao) [inline pkg/kubelet/config/file_linux_test.go]: and here.

File: pkg/kubelet/config/file_linux_test.go [hunk 5]
```diff
@@ -411,13 +411,11 @@ func writeFile(filename string, data []byte) error {
...
 	return err
```
Comment (xiangpengzhao) [inline pkg/kubelet/config/file_linux_test.go]: I'd prefer the logic like this:
```go
 func writeFile(filename string, data []byte) error {
 	f, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE, 0666)
  	if err != nil {
  		return err
  	}
	defer func() {
                err1 := f.Close()
		if err == nil {
			err = err1
		}
	}()
 	n, err := f.Write(data)
	if err != nil {
		return err
	}
  	if n < len(data) {
  		err = io.ErrShortWrite
  	}
  	return err
  }






```

**Review by xiangpengzhao — CHANGES_REQUESTED** [review]

File: pkg/kubelet/config/file_linux_test.go (hunk 3 above)
Comment (zhangxiaoyu-zidif) [inline pkg/kubelet/config/file_linux_test.go]: derekwaynecarr thinks it is redundant, it is impossible to occur error.

File: pkg/kubelet/config/file_linux_test.go (hunk 5 above)
Comment (zhangxiaoyu-zidif) [inline pkg/kubelet/config/file_linux_test.go]: well, let me see, and do some amend~

**Pull request closed without merging**